rundeckapp/grails-app/services/rundeck/services/asyncimport/AsyncImportService.groovy

```diff
@@ -0,0 +1,927 @@
+                    it.lastUpdate = AsyncImportMilestone.M1_CREATED.name
+                    return it
+                }
+            }
+            def jsonStatus = statusPersist as JSON
+            def inputStream = new ByteArrayInputStream(jsonStatus.toString().getBytes(StandardCharsets.UTF_8));
+            if (!statusPersist.projectName || statusPersist.projectName.size() <= 0) {
+                log.error("No project name provided in new status.")
+                throw new MissingPropertyException("No project name provided in new status.")
+            }
+            final def fwkProject = frameworkService.getFrameworkProject(statusPersist.projectName)
+            final def filename = JSON_FILE_PREFIX + statusPersist.projectName + JSON_FILE_EXT
+            resource = fwkProject.storeFileResource(filename, inputStream)
+            inputStream.close();
+            return resource
+        } catch (IOException e) {
+            e.printStackTrace();
+            throw e
+        }
+    }
+
+    /**
+     * async import status file updater, it replace the non-null attributes of "newStatus" object in the
+     * existing status file stored in db, leaving the null ones in the same object as they are in db.
+     *
+     * @param newStatus
+     * @return Long (bytes written)
+     */
+    def asyncImportStatusFileUpdater( AsyncImportStatusDTO newStatus, ByteArrayOutputStream out = null ){
+        if( out == null ){
+            out = new ByteArrayOutputStream()
+        }
+        def oldStatus = getAsyncImportStatusForProject(newStatus.projectName, out)
+        if( oldStatus == null ) throw new AsyncImportException("No status file for project: ${newStatus.projectName}")
+        if( oldStatus.errors != null ){
+            def oldStatusErrors = oldStatus.errors
+            newStatus.errors = oldStatusErrors + ", " + newStatus.errors
+        }
+        AsyncImportStatusDTO.replacePropsInTargetDTO(newStatus, oldStatus)
+        return saveAsyncImportStatusForProject(null, newStatus)
+    }
+
+    /**
+     * Async import process Milestone 1: Transaction Requirements.
+     * This method is synchronous, it validates the requirements to start the whole operation:
+     *
+     * a) Upload archive copy in /tmp without errors (rundeck server)
+     * b) Working dir creation, which is the target dir in which all the executions and required files will be
+     * moved to be worked on later
+     * c) Creation of the "model project", a project in wich every single executions partition will be copied and
+     * then zipped to be sent to "import to project" method.
+     *
+     * @param projectName - String
+     * @param authContext - AuthContext
+     * @param project - IRundeckProject
+     * @param inputStream - InputStream (zipped project)
+     * @param options - Import params
+     * @return result - Import results
+     */
+    @Subscriber(AsyncImportEvents.ASYNC_IMPORT_EVENT_MILESTONE_1)
```

Comment:
check if this method can be separated into multiple methods for better reading and easier testing